net/portmapper/portmapper.go

```diff
@@ -62,6 +62,8 @@ type Client struct {
 
 	localPort  uint16
 	pmpMapping *pmpMapping // non-nil if we have a PMP mapping
+
+	*Prober // non-nil once the prober has started
```

Comment:
did you mean to embed this? mean to export it? is it guarded by the mutex? looks like it is by position in the struct, but the caller doesn't know that and doesn't need to obey, as the mutex is unexported (and shouldn't be exported)

Reply:
hmmm yea I was hoping to set it once then make it so that it's never unset, so a lock wouldn't be needed. I changed it so it's guarded by the mutex in the `Probe` function, `NewProber -> newProber`.